politeiawww/database/localdb/localdb.go

```diff
@@ -260,6 +267,169 @@ func (l *localdb) AllUsers(callbackFn func(u *database.User)) error {
 	return iter.Error()
 }
 
+// NotificationNew adds a notification into a user's mailbox
+func (l *localdb) NotificationNew(n database.Notification, email string) error {
+	l.Lock()
+	defer l.Unlock()
+
+	if l.shutdown {
+		return database.ErrShutdown
+	}
+
+	log.Debugf("NotificationNew\n")
+
+	// Make sure user exists
+	exists, err := l.userdb.Has([]byte(email), nil)
+	if err != nil {
+		return err
+	} else if !exists {
+		return database.ErrUserNotFound
+	}
+
+	key := getUserNotificationsKey(email)
+	var notifications []database.Notification
+
+	n.Viewed = false
+	n.Timestamp = time.Now().Unix()
+	// Check if the user has already a mailbox registered
+	exists, err = l.userdb.Has(key, nil)
```

Comment:
Couldn't we store the notifications themselves within the user object?

Reply:
We can actually. The way I did was totally an implementation option in order to have a flatter design but on the other hand, having it within the user object would reduce the number of database queries.

Reply:
Yeah that's true, if we migrate to sql then we would use a separate table.